### PR TITLE
docs: restructure local-dev and rename to tooling

### PR DESCRIPTION
**Pull request by Rugvip**
Thoughts on this restructure? I though it might be a bit clearer. Moved it down as well since I don't think it needs to be as far up.

image

<br>
<br>

Other names considered:
  - Dev Tooling
  - Development Tools
  - CLI Tools

### CODE REVIEW

File: microsite/sidebars.json [hunk 1]
```diff
@@ -358,6 +345,25 @@
         ]
       }
     ],
+    "Tooling": [
```
Comment (camilaibs) [inline microsite/sidebars.json:348]: What about?

```
Local development
├── Tools
    └──  CLI
├── Guides
    ├── Debugging
    └── Linking local packages
```

File: .changeset/beige-eagles-cheat.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli-node': patch
+---
+
+Updated doc link.
```
Comment (camilaibs) [inline .changeset/beige-eagles-cheat.md:5]: Do we need setup redirects from the old urls to the new ones to avoid broken links?

**Review by camilaibs — COMMENTED** [review]
This is nice, left a minor comments 🎉

File: microsite/sidebars.json (hunk 1 above)
Comment (Rugvip) [reply microsite/sidebars.json:348]: The reasoning for renaming from "Local Development" was that it's not just local development, it's CI and production builds too

File: .changeset/beige-eagles-cheat.md (hunk 2 above)
Comment (Rugvip) [reply .changeset/beige-eagles-cheat.md:5]: We done have short links for some bits, but didn't bother with this one ofc. I'd say it's fine to have this one be broken for a lil bit

**Review by aramissennyeydd — APPROVED** [review]

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**